xmake.lua

```diff
@@ -1,4 +1,4 @@
-set_languages("cxx17")
+set_languages("cxx20")
```

Comment:
This breaks visual studio projects afaik

Reply:
Solution was created with no issues with this set as is here, just checked by following steps in BUILDING.md.

It resolves compiler warning, that is why I changed it.
As said, just tested that it works. Here is output from build without this and after that with it.
All is fine and dandy :)

Output without change from building generated solution:
```
Build started...
1>------ Build started: Project: RED4ext.SDK, Configuration: debug x64 ------
2>------ Build started: Project: cyber_engine_tweaks, Configuration: debug x64 ------
2>$xmake config -P . -p windows -m debug -a x64 -o "build"
1>$xmake config -P . -p windows -m debug -a x64 -o "build"
2>checking for Microsoft Visual Studio (x64) version ... 2019
1>$xmake build -P . -w RED4ext.SDK
2>$xmake build -P . -w cyber_engine_tweaks
1>[ 18%]: compiling.debug vendor\RED4ext.SDK\src\CName.cpp
1>[ 72%]: compiling.debug vendor\RED4ext.SDK\src\Scripting\Stack.cpp
1>[ 45%]: compiling.debug vendor\RED4ext.SDK\src\RTTISystem.cpp
1>[ 81%]: compiling.debug vendor\RED4ext.SDK\src\Scripting\Utils.cpp
1>[ 27%]: compiling.debug vendor\RED4ext.SDK\src\CString.cpp
1>[ 36%]: compiling.debug vendor\RED4ext.SDK\src\GameEngine.cpp
1>[ 54%]: compiling.debug vendor\RED4ext.SDK\src\RTTITypes.cpp
1>[ 63%]: compiling.debug vendor\RED4ext.SDK\src\Scripting\Functions.cpp
1>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
1>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
1>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
1>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
1>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
1>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
1>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
1>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
1>[ 90%]: archiving.debug RED4ext.SDK.lib
1>[100%]: build ok!
1>Done building project "RED4ext.SDK.vcxproj".
2>[  4%]: compiling.debug src\stdafx.h
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>[ 24%]: compiling.debug src\dllmain.cpp
2>[ 26%]: compiling.debug src\Image.cpp
2>[ 28%]: compiling.debug src\kiero\kiero.cpp
2>[ 31%]: compiling.debug src\Options.cpp
2>[ 33%]: compiling.debug src\overlay\imgui_impl_dx12.cpp
2>[ 35%]: compiling.debug src\overlay\imgui_impl_win32.cpp
2>[ 37%]: compiling.debug src\overlay\Overlay.cpp
2>[ 40%]: compiling.debug src\overlay\Overlay_D3D12.cpp
2>[ 42%]: compiling.debug src\overlay\Overlay_Hooks.cpp
2>[ 44%]: compiling.debug src\patches\Avx.cpp
2>[ 46%]: compiling.debug src\patches\DisableBoundaries.cpp
2>[ 48%]: compiling.debug src\patches\DisableIntroMovies.cpp
2>[ 51%]: compiling.debug src\patches\DisableVignette.cpp
2>[ 53%]: compiling.debug src\patches\EnableDebug.cpp
2>[ 55%]: compiling.debug src\patches\OptionsPatch.cpp
2>[ 57%]: compiling.debug src\patches\RemovePeds.cpp
2>[ 60%]: compiling.debug src\patches\SkipStartScreen.cpp
2>[ 62%]: compiling.debug src\patches\Smt.cpp
2>[ 64%]: compiling.debug src\patches\VirtualInput.cpp
2>[ 66%]: compiling.debug src\Pattern.cpp
2>[ 68%]: compiling.debug src\reverse\BasicTypes.cpp
2>[ 71%]: compiling.debug src\reverse\Converter.cpp
2>[ 73%]: compiling.debug src\reverse\SingletonReference.cpp
2>[ 75%]: compiling.debug src\reverse\StrongReference.cpp
2>[ 77%]: compiling.debug src\reverse\Type.cpp
2>[ 80%]: compiling.debug src\reverse\WeakReference.cpp
2>[ 82%]: compiling.debug src\scripting\GameOptions.cpp
2>[ 84%]: compiling.debug src\scripting\ScriptContext.cpp
2>[ 86%]: compiling.debug src\scripting\Scripting.cpp
2>[ 88%]: compiling.debug src\scripting\ScriptStore.cpp
2>[ 91%]: compiling.debug src\Utils.cpp
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>cl : Command line warning D9025: overriding '/std:c++17' with '/std:c++latest'
2>[ 97%]: linking.debug cyber_engine_tweaks.asi
2>[100%]: build ok!
2>Done building project "cyber_engine_tweaks.vcxproj".
========== Build: 2 succeeded, 0 failed, 0 up-to-date, 0 skipped ==========
```

Output without change from building generated solution:
```
Rebuild started...
1>------ Rebuild All started: Project: RED4ext.SDK, Configuration: debug x64 ------
2>------ Rebuild All started: Project: cyber_engine_tweaks, Configuration: debug x64 ------
1>$xmake clean -P . -a RED4ext.SDK
2>$xmake clean -P . -a cyber_engine_tweaks
1>$xmake config -P . -p windows -m debug -a x64 -o "build"
2>checking for architecture ... x64
2>checking for Microsoft Visual Studio (x64) version ... 2019
2>$xmake config -P . -p windows -m debug -a x64 -o "build"
1>checking for Microsoft Visual Studio (x64) version ... 2019
2>$xmake build -P . -w cyber_engine_tweaks
1>$xmake build -P . -w RED4ext.SDK
2>[  4%]: compiling.debug src\stdafx.h
2>[  6%]: compiling.debug vendor\RED4ext.SDK\src\CName.cpp
2>[  8%]: compiling.debug vendor\RED4ext.SDK\src\CString.cpp
2>[ 11%]: compiling.debug vendor\RED4ext.SDK\src\GameEngine.cpp
2>[ 13%]: compiling.debug vendor\RED4ext.SDK\src\RTTISystem.cpp
2>[ 15%]: compiling.debug vendor\RED4ext.SDK\src\RTTITypes.cpp
2>[ 17%]: compiling.debug vendor\RED4ext.SDK\src\Scripting\Functions.cpp
2>[ 20%]: compiling.debug vendor\RED4ext.SDK\src\Scripting\Stack.cpp
2>[ 22%]: compiling.debug vendor\RED4ext.SDK\src\Scripting\Utils.cpp
2>[ 24%]: compiling.debug src\dllmain.cpp
2>[ 26%]: compiling.debug src\Image.cpp
2>[ 28%]: compiling.debug src\kiero\kiero.cpp
2>[ 31%]: compiling.debug src\Options.cpp
2>[ 33%]: compiling.debug src\overlay\imgui_impl_dx12.cpp
2>[ 35%]: compiling.debug src\overlay\imgui_impl_win32.cpp
2>[ 37%]: compiling.debug src\overlay\Overlay.cpp
2>[ 40%]: compiling.debug src\overlay\Overlay_D3D12.cpp
2>[ 42%]: compiling.debug src\overlay\Overlay_Hooks.cpp
2>[ 44%]: compiling.debug src\patches\Avx.cpp
2>[ 46%]: compiling.debug src\patches\DisableBoundaries.cpp
2>[ 48%]: compiling.debug src\patches\DisableIntroMovies.cpp
2>[ 51%]: compiling.debug src\patches\DisableVignette.cpp
2>[ 53%]: compiling.debug src\patches\EnableDebug.cpp
2>[ 55%]: compiling.debug src\patches\OptionsPatch.cpp
2>[ 57%]: compiling.debug src\patches\RemovePeds.cpp
2>[ 60%]: compiling.debug src\patches\SkipStartScreen.cpp
2>[ 62%]: compiling.debug src\patches\Smt.cpp
2>[ 64%]: compiling.debug src\patches\VirtualInput.cpp
2>[ 66%]: compiling.debug src\Pattern.cpp
2>[ 68%]: compiling.debug src\reverse\BasicTypes.cpp
2>[ 71%]: compiling.debug src\reverse\Converter.cpp
2>[ 73%]: compiling.debug src\reverse\SingletonReference.cpp
2>[ 75%]: compiling.debug src\reverse\StrongReference.cpp
2>[ 77%]: compiling.debug src\reverse\Type.cpp
2>[ 80%]: compiling.debug src\reverse\WeakReference.cpp
2>[ 82%]: compiling.debug src\scripting\GameOptions.cpp
2>[ 84%]: compiling.debug src\scripting\ScriptContext.cpp
2>[ 86%]: compiling.debug src\scripting\Scripting.cpp
2>[ 88%]: compiling.debug src\scripting\ScriptStore.cpp
2>[ 91%]: compiling.debug src\Utils.cpp
2>[ 93%]: archiving.debug RED4ext.SDK.lib
2>[ 97%]: linking.debug cyber_engine_tweaks.asi
2>[100%]: build ok!
1>[100%]: build ok!
========== Rebuild All: 2 succeeded, 0 failed, 0 skipped ==========
```

Reply:
Yes but intelisense doesn't work with C++20 yet